### ISSUE TITLE
`dart format` does not automatically pick up `enable-experiment` even though the output says that it does

### ISSUE THREAD

**Issue by incendial** [dart-lang/dart_style#1886]
Hello,

When running `dart format` on a file with primary constructors, it outputs:

```text
Could not format because the source could not be parsed:

line 190, column 12 of lib/src/analyzers/lint_analyzer/rules/newline_before_constructor_rule.dart: This requires the 'primary-constructors' language feature to be enabled.
    ╷
190 │ class Point(final int x, final int y);
    │            ^
    ╵
line 190, column 38 of lib/src/analyzers/lint_analyzer/rules/newline_before_constructor_rule.dart: This requires the 'primary-constructors' language feature to be enabled.
    ╷
190 │ class Point(final int x, final int y);
    │                                      ^
    ╵
```

even though the feature is enabled in the `analysis_options.yaml` file:

```yaml
analyzer:
  enable-experiment:
    - primary-constructors
```

When running the command with `--enable-experiment` (`dart format --enable-experiment=primary-constructors`), it says:

```text
'primary-constructors' is now enabled by default; this flag is no longer required.
```

which is confusing.

It either should not mention that the flag is no longer required or pick up the `enable-experiment` config automatically.

SDK constraints in `pubspec.yaml`: ">=3.12.0 <4.0.0"
Dart SDK version: 3.14.0-65.0.dev (dev) (Fri Jul 24 13:06:40 2026 -0700) on "macos_arm64"

**Comment by FMorschel** [comment]
Somewhat related to:

- https://github.com/dart-lang/sdk/issues/63111

CC @munificent

**Comment by munificent** [comment]
I need to investigate more, but I suspect that the analysis server in the IDE plumbs experiment flags from the "analysis_options.yaml" file through to dart_style's library API as it should, but the command line `dart format` doesn't do the same.

I'm in the middle of rewriting a bunch of that code, so I'll leave this open for now until I get a chance to take a look.
